spec/components/portfolios/details_component_spec.rb

```diff
@@ -107,5 +126,36 @@ def render_component(...)
         expect(subject).to have_test_selector("op-portfolios--updated-at", text: "Updated about 1 month ago")
       end
     end
+
+    describe "sub-item status" do
+      context "when none of the sub-items has a status" do
+        let(:status_code_a) { nil }
+        let(:status_code_b) { nil }
+
+        it "does not render a progress bar" do
+          expect(subject).not_to have_test_selector("op-portfolios--sub-status-bar")
+        end
+      end
+
+      context "when some of the sub-items have a status set" do
+        let(:status_code_b) { nil }
+
+        it "renders a progress bar detailing the status of child programs and projects" do
+          expect(subject).to have_test_selector("op-portfolios--sub-status-bar")
+
+          expect(subject).to have_test_selector("op-portfolios--status-#{status_code_a}")
+          expect(subject).not_to have_test_selector("op-portfolios--status-#{status_code_b}")
+        end
+      end
+
+      context "when all of the sub-items have a status set" do
+        it "renders a progress bar detailing the status of child programs and projects" do
+          expect(subject).to have_test_selector("op-portfolios--sub-status-bar")
+
+          expect(subject).to have_test_selector("op-portfolios--status-#{status_code_a}")
+          expect(subject).to have_test_selector("op-portfolios--status-#{status_code_b}")
```

Comment:
```suggestion
          expect(subject).to have_test_selector("op-portfolios--status-#{status_code_a}")
          expect(subject).to have_test_selector("op-portfolios--status-#{status_code_b}")
          expect(subject).to have_no_test_selector("op-portfolios--status-not_set")
```
Not sure if it better to stick with `.to have_no_test_selector` to get used to using only that, or use `.not_to have_test_selector`, as it is not a feature spec